js/src/Zone.js

```diff
@@ -0,0 +1,84 @@
+import ExpectedMark from "./ExpectedMark";
+import UXCapture from "./UXCapture";
+
+/**
+ * Zone represents collection of elements groupped together and corresponding phase of page load
+ *
+ * Only one set of zones can be tracked for the same view at one point in time
+ */
+export default class Zone {
+  // constructs individual zone object
+  constructor(config) {
+    // {
+    //   name: "ux-destination-verified",
+    //   marks: ["ux-image-online-logo", "ux-image-inline-logo"]
+    //   selectors: ["img.logo"],
+    //   startMark: "navigationStart",
+    //   onMeasure: measureName => {}
+    //   onMark: markName => {}
+    // }
+
+    // name to be used for UserTiming measures
+    this.measureName = config.name;
+
+    // callback to execute when Zone is complete
+    this.onMeasure = config.onMeasure;
+
+    // callback for marks to call when they are complete
+    this.onMark = config.onMark;
+
+    // array of mark names that were recorded so far
+    this.recordedMarkNames = [];
+
+    this.startMarkName = config.startMarkName || "navigationStart";
+
+    // look up existing mark object or create a new one
+
+    // now just string names for UserTiming marks
+    // in the future, different types of events, e.g. ImageElement("logo"), PaintTimer("first-paint")
+    this.marks = config.marks.map(markName => {
+      const mark = ExpectedMark.get(markName) || new ExpectedMark(markName);
```

Comment:
if we had a `ExpectedMark.create` static method, that could handle this `||` logic internally